### PR TITLE
Added fixes for some build errors and support for the e2k architecture

### PR DESCRIPTION
**Pull request by ArtemS2**
Through joint efforts, we were able to add initial support for the e2k architecture. During the assembly process, the following errors were found and described ways to fix them.

Ref: https://lvee.org/en/abstracts/303

### CODE REVIEW

**Review by r-a-sattarov — APPROVED** [review]

**Review by reuksv — APPROVED** [review]

File: Q3E/src/main/jni/doom3/neo/quake4/ai/AI_Move.cpp [hunk 1]
```diff
@@ -3889,10 +3889,10 @@ class rvPathFinder {
 
 	/*
 	=====================
-	Success
+	success
```
Comment (Toxblh) [inline Q3E/src/main/jni/doom3/neo/quake4/ai/AI_Move.cpp:3892]: That and other changes with Success -> success is it really need? Looks like a mistake refactor in the file. Other bool functions stayed as it

**Review by Toxblh — COMMENTED** [review]
Req more info regarding success() function

File: Q3E/src/main/jni/doom3/neo/quake4/ai/AI_Move.cpp (hunk 1 above)
Comment (Toxblh) [reply Q3E/src/main/jni/doom3/neo/quake4/ai/AI_Move.cpp:3892]: I see, that is X11 fix

**Review by Toxblh — APPROVED** [review]

**Comment by ilyakurdyukov** [comment]
Whitelisting architectures with 64-bit pointers is a terrible idea.

`#if defined(__aarch64__) || defined(__x86_64__) || defined(__e2k__)`

Use `__LP64__` for this.

**Comment by r-a-sattarov** [comment]
Thanks @ArtemS2. The game is running quite well on [Elbrus-8C](https://en.wikipedia.org/wiki/Elbrus-8S)

- https://www.youtube.com/watch?v=jnD8TX6gmEA

File: Q3E/src/main/jni/doom3/neo/quake4/ai/AI_Move.cpp (hunk 1 above)
Comment (glKarin) [reply Q3E/src/main/jni/doom3/neo/quake4/ai/AI_Move.cpp:3892]: I removed include path `externlibs/include`, this headers only used in MinGW or MSVC, so don't need in Linux. so don't change function name of `Success`.

**Pull request merged**